x-pack/plugins/serverless_search/server/collectors/connectors/telemetry.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IClusterClient, Logger } from '@kbn/core/server';
+
+import { CONNECTORS_INDEX } from '@kbn/search-connectors';
+import { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+
+interface Telemetry {
+  native: {
+    total: number;
+  };
+  clients: {
+    total: number;
+  };
+}
+
+/**
+ * Register the telemetry collector
+ */
+
+export const registerTelemetryUsageCollector = (
+  usageCollection: UsageCollectionSetup,
+  client: IClusterClient,
+  log: Logger
+) => {
+  const telemetryUsageCollector = usageCollection.makeUsageCollector<Telemetry>({
+    type: 'connectors_serverless',
+    fetch: async () => fetchTelemetryMetrics(client, log),
+    isReady: () => true,
+    schema: {
+      native: {
+        total: { type: 'long' },
+      },
+      clients: {
+        total: { type: 'long' },
+      },
+    },
+  });
+  usageCollection.registerCollector(telemetryUsageCollector);
+};
+
+/**
+ * Fetch the aggregated telemetry metrics
+ */
+
+export const fetchTelemetryMetrics = async (
+  client: IClusterClient,
+  log: Logger
```

Comment:
Is that an exact duplicate from the collector in `x-pack/plugins/enterprise_search/server/collectors/connectors/telemetry.ts`? if so, we might want to factorize this in a package or something?

Reply:
It was in a package, and removed by this commit: https://github.com/elastic/kibana/pull/172280/commits/eb0eeafb7d1a3aa76c23902f0a93018c66d03082

Because [CI](https://buildkite.com/elastic/kibana-pull-request/builds/180495#018c2081-9fa9-4ef5-a07d-d89efaed4242) failed with
```
[14:05:33] → Error extracting collector in x-pack/plugins/enterprise_search/server/collectors/connectors/telemetry.ts
Error: Unable to find identifier in source Telemetry
ERROR Unhandled exception!
ERROR Error: Error extracting collector in x-pack/plugins/enterprise_search/server/collectors/connectors/telemetry.ts
      Error: Unable to find identifier in source Telemetry
          at createFailError (dev_cli_errors.ts:27:24)
          at parseUsageCollection (ts_parser.ts:226:32)
          at parseUsageCollection.next (<anonymous>)
          at extractCollectors (extract_collectors.ts:58:32)
          at extractCollectors.next (<anonymous>)
          at Task.task (extract_collectors_task.ts:43:53)
          at processTicksAndRejections (node:internal/process/task_queues:95:5)
🚨 Error: The command exited with status 1
```

Reply:
Thanks for explaining.

@Bamieh I thought we addressed the problems with telemetry schemas in packages?

Reply:
We did but I think this is just failing to parse the interface for some reason, i'm pulling the code to debug what is happening

Reply:
The `Telemetry` interface was defined in `package` at that time. Maybe this is the reason?

Reply:
I created an issue to track progress and any PR fixes needed for this. https://github.com/elastic/kibana/issues/172577

I dont think it should block this PR if we are able to merge it soon I recommend we can do the refactoring after in a separate PR. @pgayvallet wdyt?

Reply:
One of the reasons I completely dropped the function in `package` is that, in the next release, the usage collection logic will be wrapped into an API, so both usage collectors in `enterprise_search` and `serverless_search` will use the new API instead of the `fetchTelemetryMetrics` defined here.

Reply:
> I dont think it should block this PR if we are able to merge it soon 

No, we don't need to block, especially on a problem coming from our side. I was just wondering, because I thought those problem were already fixed.